test/helpers/foundry/foundryup.ts

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env -S node --require "./node_modules/tsx/dist/preflight.cjs" --import "./node_modules/tsx/dist/loader.mjs"
+
+import { join, relative } from 'node:path';
+import { homedir } from 'node:os';
+import { Dir, readFileSync } from 'node:fs';
+import { copyFile, opendir, rm, symlink, unlink } from 'node:fs/promises';
+import { createHash } from 'node:crypto';
+import { cwd, exit } from 'node:process';
+import { parse as parseYaml } from 'yaml';
+import {
+  getVersion,
+  isCodedError,
+  noop,
+  say,
+  transformChecksums,
+} from './utils';
+import { extractFrom } from './extract';
+import { parseArgs, printBanner } from './options';
+import { Architecture, Checksums, Extension, Platform } from './types';
+
+export function getCacheDirectory(): string {
+  let enableGlobalCache = false;
+  try {
+    const configFileContent = readFileSync('.yarnrc.yml', 'utf8');
+    const parsedConfig = parseYaml(configFileContent);
+    enableGlobalCache = parsedConfig?.enableGlobalCache ?? false;
+  } catch (error) {
+    console.error('Error reading/parsing .yarnrc.yml:', error);
+  }
+  return enableGlobalCache
+    ? join(homedir(), '.cache', 'metamask')
+    : join(cwd(), '.metamask', 'cache');
```

Comment:
Probably not necessary, but this ensures that we keep our own cache alongside the `.yarn` cache. Why? I had to put it somewhere, and copying whatever we do for yarn just felt right, I guess.